jenkins/deployment/community/cortx-community-deploy.groovy

```diff
@@ -15,156 +14,170 @@ pipeline {
-                    echo key_name = '"'$KEY_NAME'"' | cat >>user.tfvars
-                    cat user.tfvars | tail -6
-                popd
+                    sed -i 's,os_version          =.*,os_version          = "'"$OS_VERSION"'",g' user.tfvars && sed -i 's,region              =.*,region              = "'"$REGION"'",g' user.tfvars && sed -i 's,security_group_cidr =.*,security_group_cidr = "'"$VM_IP/32"'",g' user.tfvars && sed -i 's,instance_count          =.*,instance_count          = "'"$INSTANCE_COUNT"'",g' user.tfvars && sed -i 's,ebs_volume_count          =.*,ebs_volume_count          = "'"$VOLUME_COUNT"'",g' user.tfvars && sed -i 's,ebs_volume_size          =.*,ebs_volume_size          = "'"$VOLUME_SIZE"'",g' user.tfvars && sed -i 's,tag_name          =.*,tag_name          = "'"$INSTANCE_TAG_NAME"'",g' user.tfvars
+                    echo key_name            = '"'$KEY_NAME'"' | cat >>user.tfvars
+                    cat user.tfvars
+                    popd
                 '''
             }
         }
-
-                stage ('Create EC2 instace') {
+        stage('Create Multi EC2 instances') {
             steps {
                 script { build_stage = env.STAGE_NAME }
-                sh label: 'Setting up EC2 instance', script: '''
+                sh label: 'Setting up multi EC2 instances', script: '''
                     pushd solutions/community-deploy/cloud/AWS
                         terraform validate && terraform apply -var-file user.tfvars --auto-approve
-                    popd
+                        popd
             '''
             }
         }
-
-        stage ('Network and storage configuration') {
+        stage('Configure network and storage') {
             steps {
                 script { build_stage = env.STAGE_NAME }
-                script {
-                    env.AWS_IP = sh( script: '''
-                    cd solutions/community-deploy/cloud/AWS && terraform show -json terraform.tfstate | jq .values.outputs.cortx_deploy_ip_addr.value | tr -d '"'
-                    ''', returnStdout: true).trim()
-                }
                 catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
-                sh label: 'Setting up Network and Storage devices for CORTX. Script will reboot the instance on completion', script: '''
-                pushd solutions/community-deploy/cloud/AWS
-                    ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@"${AWS_IP}" sudo bash /home/centos/setup.sh
-                    sleep 240
-                popd
-                '''
+                    sh label: 'Setting up Network and Storage devices for CORTX. Script will reboot the instance on completion', script: '''
+                    pushd solutions/community-deploy/cloud/AWS
+                        export PUBLIC_IP=$(terraform show -json terraform.tfstate | jq .values.outputs.aws_instance_public_ip_addr.value 2>&1 | tee ip_public.txt | tr -d '",[]' | sed '/^$/d')
```

Comment:
Can we create the Jenkins env variable so we do not need to fetch it from `ip_public.txt` at each stage.  Refer - https://github.com/Seagate/cortx-re/blob/main/jenkins/deployment/community/cortx-community-deploy.groovy#L81

Reply:
lets take this as code improvement